pkg/devfile/image/image.go

```diff
@@ -91,12 +102,9 @@ func buildPushImage(backend Backend, fs filesystem.Filesystem, image *devfile.Im
 // selectBackend selects the container backend to use for building and pushing images
 // It will detect podman and docker CLIs (in this order),
 // or return an error if none are present locally
-func selectBackend() (Backend, error) {
+func selectBackend(ctx context.Context) (Backend, error) {
 
-	podmanCmd := getEnvFunc("PODMAN_CMD")
-	if podmanCmd == "" {
-		podmanCmd = "podman"
-	}
+	podmanCmd := pointer.StringDeref(envcontext.GetEnvConfig(ctx).PodmanCmd, "podman")
```

Comment:
Wouldn't make it more sense to set default values using struct tags in Configuration (`pkg/config/config.go`)? I can see that this is possible via the [`default` struct tag](https://github.com/kelseyhightower/envconfig#struct-tag-support).
This way, callers would not need to determine the default value.
And our future selves would have a single place to look for env vars and their default values.